### ISSUE TITLE
circleci: Bad CPU type in executable

### ISSUE THREAD

**Issue by carlos-santos-anchor** [AleksaC/circleci-cli-py#12]
Hello 👋 

I'm having a weird issue using `pre-commit` and this validator, and I' not sure if it's a mis-config or something else.

```bash
CircleCI Config Validation..................................................Failed
- hook id: circle-ci-validator
- exit code: 126

env: circleci: Bad CPU type in executable
```

Looking into the `exit code: 126` it looks like the executable does not have execution permissions.

But I'm not sure what have changed.

Does anyone have an idea of what could be the root cause?

Thanks.

----

Here's my config of `.pre-commit-config.yaml`:

```yaml
  - repo: https://github.com/AleksaC/circleci-cli-py
    rev: b6ca7f04421b627192e232b22ffc0f33e9571700  # frozen: v0.1.30549
    hooks:
      - id: circle-ci-validator
```

I'm using the latest released tag [v0.1.30549](https://github.com/AleksaC/circleci-cli-py/releases/tag/v0.1.30549)

The same happens with the [latest commit](https://github.com/AleksaC/circleci-cli-py/commit/2fb06891ae03bf9fb42357fd2dc6317b37c35946)

**Comment by AleksaC** [comment]
Hi @carlos-santos-anchor, thanks for reporting the issue!

I'm guessing you're using apple silicon. You need Rosetta to make it work since it's using amd64 binaries. I would like to add support for arm processors, but I'm not sure when will I have time to do so.

**Comment by carlos-santos-anchor** [comment]
That makes sense @AleksaC . I didn't even thought about that 😓 

From what I'm seeing in the code, this doesn't look like a huge effort, maybe I could try it out one of these days.

Confirm me if this would be enough:

We're using [`circleci-cli_0.1.30549_darwin_amd64.tar.gz`](https://github.com/CircleCI-Public/circleci-cli/releases/download/v0.1.30549/circleci-cli_0.1.30549_darwin_amd64.tar.gz) but we should be using [`circleci-cli_0.1.30549_darwin_arm64.tar.gz`](https://github.com/CircleCI-Public/circleci-cli/releases/download/v0.1.30549/circleci-cli_0.1.30549_darwin_arm64.tar.gz).

So looking into the code, [these are the images we're using](https://github.com/AleksaC/circleci-cli-py/blob/master/setup.py#L25-L27), and we would need to do a slight change to, not just compare the `sys.platform`, but also which processor.

Would `platform.processor()` work? Mixed with `platform.system()`?

e.g. running on arm M3:

```python
>>> platform.processor()
'arm'
>>> platform.system()
'Darwin'
>>>
```

also in linux:

```python
>>> platform.processor()
'x86_64'
>>> platform.system()
'Linux'
```

And then we could have something like this:

```diff
ARCHIVE_SHA256 = {
    "linux": (
        "circleci-cli_0.1.30549_linux_amd64.tar.gz",
        "807c98f96a54b2d63e54a068e215eac2af9c529f59529ad6474c4d73d8caf93c",
    ),
    "darwin-x86_64": (
        "circleci-cli_0.1.30549_darwin_amd64.tar.gz",
        "924e3f3cd375902cb564f9becebe6e98faaab319456375214885b44150628779",
    ),
+    "darwin-arm": (
+        "circleci-cli_0.1.30549_darwin_arm64.tar.gz",
+        "15a0ee5c8578617f160dab264c5cb7baeb7d5869ea30b49a02805c083b2afa97",
+    ),
    "win32": (
        "circleci-cli_0.1.30549_windows_amd64.zip",
        "0c554a10d3d8cb7a4d330c6d4f9a0b218e830dd45175d8f2d743abd338842364",
    ),
}
```

Of course, also adding the ARM variants for Linux

**Comment by AleksaC** [comment]
Yep, I used `platform.machine()` in similar projects. I don't remember why, so I compared the two on a couple machines I have access to. On one of my linux servers I get:

```
>>> platform.machine()
'x86_64'
```

vs

```
>>> platform.processor()
''
```

So I'm guessing that `platform.machine` is more reliable.

Either way if you get a chance to submit a PR I'd be glad to merge it.

**Comment by AleksaC** [comment]
The issue has been fixed in #13. I completely reworked the way new versions are generated to match my other similar projects. There is no tag for the new version but you can use a commit sha, like so:

```yaml
  - repo: https://github.com/AleksaC/circleci-cli-py
    rev: bc1d89a
    hooks:
      - id: circle-ci-validator
```

**Comment by carlos-santos-anchor** [comment]
Thanks @AleksaC , this was on my TODO list but you were faster 😢 

```python
{
    'windows_amd64': ('circleci-cli_0.1.30549_windows_amd64.zip', '0c554a10d3d8cb7a4d330c6d4f9a0b218e830dd45175d8f2d743abd338842364'), 
    'windows_arm64': ('circleci-cli_0.1.30549_windows_arm64.zip', '15a0ee5c8578617f160dab264c5cb7baeb7d5869ea30b49a02805c083b2afa97'), 
    'linux_arm64': ('circleci-cli_0.1.30549_linux_arm64.tar.gz', '649496a32305c539df9c66f737f74a8852dd73e53c14e7d95b77c0e2d7790d38'), 
    'darwin_arm64': ('circleci-cli_0.1.30549_darwin_arm64.tar.gz', '692236f5aaf7f62ebc5fc48fb5dfb55a412e8aa4361c8ba65f0f92a459002c1d'), 
    'linux_amd64': ('circleci-cli_0.1.30549_linux_amd64.tar.gz', '807c98f96a54b2d63e54a068e215eac2af9c529f59529ad6474c4d73d8caf93c'), 
    'darwin_amd64': ('circleci-cli_0.1.30549_darwin_amd64.tar.gz', '924e3f3cd375902cb564f9becebe6e98faaab319456375214885b44150628779')
}
```

```python
    archive, sha256 = ARCHIVE_SHA256[f"{os}_{arch}64"]
```

You simplified this and it looks awesome 👌 

Thank you so much, I'll give it a spin with that revision 🙏
